src/mentionui.js

```diff
@@ -254,15 +305,39 @@ export default class MentionUI extends Plugin {
 	/**
 	 * Returns a promise that resolves with autocomplete items for a given text.
 	 *
+	 * **Note**: This method unifies the responses from feed callbacks to always return a promise.
+	 *
 	 * @param {String} marker
 	 * @param {String} feedText
-	 * @return {Promise<module:mention/mention~MentionFeedItem>}
+	 * @returns {Promise<module:mention/mention~MentionFeedItem>}
 	 * @private
 	 */
 	_getFeed( marker, feedText ) {
 		const { feedCallback } = this._mentionsConfigurations.get( marker );
 
-		return Promise.resolve().then( () => feedCallback( feedText ) );
+		this._lastRequested = feedText;
+
+		// To unify later processing the _getFeed should return a promise even if feed callback returns an array.
+		return new Promise( ( resolve, reject ) => {
+			const response = feedCallback( feedText );
+
+			const isAsynchronous = response instanceof Promise;
+
+			if ( !isAsynchronous ) {
+				resolve( response );
```

Comment:
You forgot about an early return here... which opens a question – how does this even work and how is it possible that it didn't yield any errors?